plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -50,13 +52,974 @@ type CiscoTelemetryMDT struct {
+	//string to uint64
+	vals := field.GetStringValue()
+	if vals != "" {
+		if val64, err := strconv.ParseUint(vals, 10, 64); err == nil {
+			return val64
+		}
+	}
+	return nil
+}
+
+//xform string to int64
+func (c *CiscoTelemetryMDT) nxosValueXformStringToInt64(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//string to int64
+	vals := field.GetStringValue()
+	if vals != "" {
+		if val64, err := strconv.ParseInt(vals, 10, 64); err == nil {
+			return val64
+		}
+	}
+	return nil
+}
+
+//auto-xform
+func (c *CiscoTelemetryMDT) nxosValueAutoXform(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//check if we want auto xformation
+	vals := field.GetStringValue()
+	if vals != "" {
+		if val64, err := strconv.ParseUint(vals, 10, 64); err == nil {
+			return val64
+		}
+		if valf, err := strconv.ParseFloat(vals, 64); err == nil {
+			return valf
+		}
+		if val64, err := strconv.ParseInt(vals, 10, 64); err == nil {
+			return val64
+		}
+	} // switch
+	return nil
+}
+
+//auto-xform float properties
+func (c *CiscoTelemetryMDT) nxosValueAutoXformFloatProp(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//check if we want auto xformation
+	vals := field.GetStringValue()
+	if vals != "" {
+		if valf, err := strconv.ParseFloat(vals, 64); err == nil {
+			return valf
+		}
+	} // switch
+	return nil
+}
+
+//xform uint64 to string
+func (c *CiscoTelemetryMDT) nxosValueXformUint64ToString(field *telemetry.TelemetryField, value interface{}) interface{} {
+	switch val := field.ValueByType.(type) {
+	case *telemetry.TelemetryField_StringValue:
+		if len(val.StringValue) > 0 {
+			return val.StringValue
+		}
+	case *telemetry.TelemetryField_Uint64Value:
+		return (strconv.FormatUint(val.Uint64Value, 10))
+	}
+	return nil
+}
+
+//Xform value field.
+func (c *CiscoTelemetryMDT) nxosValueXform(field *telemetry.TelemetryField, value interface{}, path string) interface{} {
+	//if isNXOS
+	if !strings.ContainsRune(path, ':') {
+		if _, ok := c.propMap[field.Name]; ok {
+			return (c.propMap[field.Name](field, value))
+		} else {
+			//check if we want auto xformation
+			if _, ok := c.propMap["auto-prop-xfromi"]; ok {
+				v1 := c.propMap["auto-prop-xfrom"](field, value)
+				if v1 != nil {
+					return v1
+				}
+			} else {
+				//Now check path based conversion.
+				//If mapping is found then do the required transformation.
+				if c.nxpathMap[path] != nil {
+					switch c.nxpathMap[path][field.Name] {
+					//Xformation supported is only from String, Uint32 and Uint64
+					case "integer":
+						switch val := field.ValueByType.(type) {
+						case *telemetry.TelemetryField_StringValue:
+							if vali, err := strconv.ParseInt(val.StringValue, 10, 32); err == nil {
+								return vali
+							}
+						case *telemetry.TelemetryField_Uint32Value:
+							return int(value.(uint32))
+						case *telemetry.TelemetryField_Uint64Value:
+							return int64(value.(uint64))
```

Comment:
Please check the type assertions before using them as otherwise telegraf will panic in case the assertion doesn't hold.